File: backend/profile-service/src/main/java/com/example/profileservice/profile/dto/ProfileDto.java
package com.example.profileservice.profile.dto;

import com.example.profileservice.profile.entity.Profile;
import com.example.profileservice.profile.entity.ProfileType;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.format.DateTimeFormatter;

@Getter
@NoArgsConstructor
public class ProfileDto {

    private Long memberId;
    private ProfileType profileType;
    private String value;
    private String createdDate;
    private String updatedDate;

    public ProfileDto(Profile profile) {
        this.memberId = profile.getMemberId();
        this.profileType = profile.getProfileType();
        this.value = profile.getValue();
        this.createdDate = profile.getCreatedDate() != null ? profile.getCreatedDate().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")) : "";
        this.updatedDate = profile.getUpdatedDate() != null ? profile.getUpdatedDate().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")) : "";
    }
}

File: backend/profile-service/src/main/kotlin/com/example/profileservice/profile/dto/ProfileDto.kt
package com.example.profileservice.profile.dto

import com.example.profileservice.profile.entity.Profile
import com.example.profileservice.profile.entity.ProfileType
import java.time.format.DateTimeFormatter

data class ProfileDto(
    var id: Long? = null,
    var memberId: Long? = null,
    var profileType: ProfileType? = null,
    var value: String? = null,
    var createdDate: String? = null,
    var updatedDate: String? = null,
) {
    constructor(profile: Profile) : this(
        id = profile.id,
        memberId = profile.memberId,
        profileType = profile.profileType,
        value = profile.value,
        createdDate = profile.createdDate?.format(FORMAT) ?: "",
        updatedDate = profile.updatedDate?.format(FORMAT) ?: "",
    )

    companion object {
        private val FORMAT: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
    }
}
